### ISSUE TITLE
Missing CORS headers in POST responses (while they are correctly present in OPTIONS responses)

### ISSUE THREAD

**Issue by sergiogiogio** [rclone/rclone#6045]
#### What is the problem you are having with rclone?
Rclone does not provide the CORS headers (Access-Control-Allow-Headers, Access-Control-Allow-Origin, Access-Control-Request-Method) in POST responses. The headers are however correctly provided in OPTIONS responses, which also shows that the command line is properly parsed.

Below is an illustration with an OPTIONS and a POST requests issues by a browser (firefox) and their associated responses from rclone

- Request1: OPTIONS
```

OPTIONS /core/stats HTTP/1.1
Host: 192.168.1.1:5572
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Access-Control-Request-Method: POST
Access-Control-Request-Headers: content-type
Referer: http://192.168.1.1/
Origin: http://192.168.1.1
Connection: keep-alive

HTTP/1.1 200 OK
Access-Control-Allow-Headers: authorization, Content-Type
Access-Control-Allow-Origin: *
Access-Control-Request-Method: POST, OPTIONS, GET, HEAD
Date: Sat, 19 Mar 2022 13:27:13 GMT
Content-Length: 0
```

- Request2: POST
Notice the missing CORS headers in the response.

```POST /core/stats HTTP/1.1
Host: 192.168.1.1:5572
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Referer: http://192.168.1.1/
Content-Type: application/json
Origin: http://192.168.1.1
Content-Length: 4
Connection: keep-alive

HTTP/1.1 401 Unauthorized
Content-Type: text/plain; charset=utf-8
Www-Authenticate: Basic realm="rclone"
X-Content-Type-Options: nosniff
Date: Sat, 19 Mar 2022 13:27:14 GMT
Content-Length: 13
```

#### What is your rclone version (output from `rclone version`)
```rclone v1.52.2_2020-06-24
- os/arch: linux/mips
- go version: go1.13.15
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
openwrt / 32bit

#### Which cloud storage system are you using? (e.g. Google Drive)
Google Drive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
rclone --rc --rc-addr=0.0.0.0:5572 -v --log-file=/www/rcloneMon/uploadPhotos1/rclone.log --rc-user=xxxx --rc-pass=xxxx --rc-allow-origin=* copy /mnt/nikon/DCIM/101D5500 gphotos:/upload
```

#### Possible cause

I suspect the issue comes form this commit https://github.com/rclone/rclone/commit/75252e4a89b723e3d024f75a705e56333bc76272 where WriteHeader function was removed from handlePost function (but preserved in handleOptions functions).

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by sergiogiogio** [comment]
After further investigation - I realized I was actually having 2 issues: authentication (rclone replying 401 ) **and** CORS (rclone not replying with CORS headers). The browser was reporting the CORS issue so I set out to investigate this one first. However it turns out from rclone perspective authentication code comes before CORS code (at least for POST - there does not seem to be authentication for OPTIONS). Once I resolved the authentication issue, the CORS issue was immediately resolved. 

Issue closed!
